hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -1440,10 +1440,11 @@ public List<HoodieColumnRangeMetadata<Comparable>> getSortedColumnStatsList(
             .collect(Collectors.toList());
       } else {
         return baseFileNameList.stream().flatMap(filename ->
-                FileFormatUtils.getInstance(HoodieFileFormat.PARQUET).readColumnStatsFromMetadata(
-                    metaClient.getStorage(),
-                    new StoragePath(FSUtils.constructAbsolutePath(metaClient.getBasePathV2(), partitionPath), filename),
-                    allColumnNameList).stream())
+                new HoodieSparkIOFactory(metaClient.getStorage()).getFileFormatUtils(HoodieFileFormat.PARQUET)
```

Comment:
here also, let;s see if we can use `HoodieIOFactory`. And would be good to create the util before stream.

Reply:
Addressed.